### ISSUE TITLE
RDFSTORE.select() does not support SPARQL PREFIXES and REGEX

### ISSUE THREAD

**Issue by cedricdcc** [vliz-be-opsci/py-RDF-store#29]
For the following test:

```
def test_select_sparql(rdf_store: RDFStore):
    SPARQL = """
      SELECT DISTINCT ?s WHERE {
      [] <http://www.w3.org/ns/dcat#resource> ?s .
      }
    """

    # TODO: test if with uri encoding it works for other sparql queries

    result = rdf_store.select(SPARQL)
    print(result)
```

I get this print statement: <rdflib.plugins.sparql.results.xmlresults.XMLResult object at 0x0000020FEDD05490>

For the same test with this sparql:
```
SPARQL = """
      PREFIX dcat: <http://www.w3.org/ns/dcat#> .
      SELECT DISTINCT ?s WHERE {
      [] <http://www.w3.org/ns/dcat#resource> ?s .
      FILTER(REGEX(STR(?s), "/publication/\\d+"))
      }
    """
```

I get (400, 'HTTP Error 400: ', None)

This is a bad request. The problem is that this is handled by rdflib

**Comment by cedricdcc** [comment]
After searching it seems that this is due to the limitations of SPARQL version that the read_uri endpoint uses of graphDB.
https://www.phind.com/search?cache=fog56qnjqorj6kfgj0dgt8d8

A possible solution could be to use a url request to the graphdb endpoint itself or bumping the base image of graphDB

**Comment by cedricdcc** [comment]
Another solution would be to add a check to see if a select query is compliant to the standard of SPARQL that is used by graphDB. For the prefixes a small function can be made to insert them into the sparql query itself.

**Comment by cedricdcc** [comment]
Another possible route is the initNS variable that can be changed https://stackoverflow.com/questions/76677670/sparql-namespace-conflict-while-querying

**Comment by marc-portier** [comment]
with the new yasgui component to kgap it shows that the 415 error pops up when performing the sparql reads (select) over the write_uri

**read_uri works**
![image](https://github.com/vliz-be-opsci/py-RDF-store/assets/299574/1fc217c1-7086-4fe7-a638-8a55d37b6290)


**write_uri triggers 415**
![image](https://github.com/vliz-be-opsci/py-RDF-store/assets/299574/dc654036-0030-4b33-913a-233885e16081)

**Comment by marc-portier** [comment]
for another issue looking into graphdb logs of running tests

and suddenly find this:
```[INFO ] 2024-03-29 15:15:19,669 [repositories/rdf_store_test | c.o.f.s.GraphDBProtocolExceptionResolver] Client sent bad request (400)
org.eclipse.rdf4j.http.server.ClientHTTPException: MALFORMED QUERY: Multiple prefix declarations for prefix 'schema'
	at org.eclipse.rdf4j.http.server.repository.handler.DefaultQueryRequestHandler.getQuery(DefaultQueryRequestHandler.java:185)
	at com.ontotext.graphdb.sesame.handler.GraphDBQueryResultHandler.getQuery(GraphDBQueryResultHandler.java:75)
	at org.eclipse.rdf4j.http.server.repository.handler.AbstractQueryRequestHandler.handleQueryRequest(AbstractQueryRequestHandler.java:70)
	at org.eclipse.rdf4j.http.server.repository.AbstractRepositoryController.handleRequestInternal(AbstractRepositoryController.java:53)
...
```

this **«Multiple prefix declarations for prefix 'schema'»** looks like somewhere in the chain something is by default adding the `schema:` prefix into the sparql-select stack 


possible candidates for that (unwanted / unexpected) injection:
- rdflib sparqupdate store, or query, ...
- graphdb itself

so it might just be so that:
- using another prefix like `schemA:` or even skip the extra declaration -- will make your issue go away instantly (arguably not a long term solution) 
- the claim of this issue is not really generally valid for all `prefix` statements -- which would explain why the current test that was created does not show the error!

**Comment by marc-portier** [comment]
rdflib is doing this see docs at https://rdflib.readthedocs.io/en/stable/namespaces_and_bindings.html#namespacemanager

we should apply `bind_namespaces= "none"` consistently accross the `store.py` to overrule the standard behaviour in rdflib

**Comment by marc-portier** [comment]
closing this for the moment, as the acknowledged partial problem this raised was solved together with #32 - and fixed in application of #34
